test/jdk/javax/swing/border/EtchedBorder/ScaledEtchedBorderTest.java

```diff
@@ -66,165 +71,256 @@ public class ScaledEtchedBorderTest {
+                                throwWrongThickness(thickness, borderThickness, x, y);
+                            }
+                            borderThickness = 1;
+                            state = State.LEFT_HIGHLIGHT;
+                        } else {
+                            throwUnexpectedColor(x, y, color);
+                        }
+                        break;
+
+                    case LEFT_HIGHLIGHT:
+                        if (color == HIGHLIGHT.getRGB()) {
+                            borderThickness++;
+                        } else if (color == INSIDE_COLOR.getRGB()) {
+                            if (borderThickness != thickness) {
+                                throwWrongThickness(thickness, borderThickness, x, y);
+                            }
+                            borderThickness = 0;
+                            state = State.INSIDE;
+                        } else {
+                            throwUnexpectedColor(x, y, color);
+                        }
+                        break;
+
+                    case INSIDE:
+                        if (color == SHADOW.getRGB()) {
+                            state = State.RIGHT_SHADOW;
+                            borderThickness = 1;
+                        } else if (color != INSIDE_COLOR.getRGB()) {
+                            throwUnexpectedColor(x, y, color);
+                        }
+                        break;
+
+                    case RIGHT_SHADOW:
+                        if (color == SHADOW.getRGB()) {
+                            borderThickness++;
+                        } else if (color == HIGHLIGHT.getRGB()) {
+                            if (borderThickness != thickness) {
+                                throwWrongThickness(thickness, borderThickness, x, y);
+                            }
+                            borderThickness = 1;
+                            state = State.RIGHT_HIGHLIGHT;
+                        } else {
+                            throwUnexpectedColor(x, y, color);
+                        }
+                        break;
+
+                    case RIGHT_HIGHLIGHT:
+                        if (color == HIGHLIGHT.getRGB()) {
+                            borderThickness++;
+                        } else if (color == OUTER_COLOR.getRGB()) {
+                            if (borderThickness != thickness) {
+                                throwWrongThickness(thickness, borderThickness, x, y);
+                            }
+                            borderThickness = 0;
+                            state = State.BACKGROUND;
+                        } else {
+                            throwUnexpectedColor(x, y, color);
+                        }
+                        break;
                 }
-            }
-        }
+            } while (yStep > 0 && ((y += yStep) < height));
+        } while (xStep > 0 && ((x += xStep) < width));
+    }
+
+    private enum State {
```

Comment:
enum declaration can be moved before its first usage (before `checkBorder()`)

Reply:
I agree here, the `State` enum is declared before `checkBorder` in `ScaledTextFieldBorderTest.java`, yet it's after the method in both `ScaledLineBorderTest.java` and `ScaledEtchedBorderTest.java`.